compat/src/main/scala-2.11/scala/util/Using.scala

```diff
@@ -0,0 +1,409 @@
+      resource2: => R2,
+      resource3: => R3
+  )(body: (R1, R2, R3) => A): A =
+    resource(resource1) { r1 =>
+      resource(resource2) { r2 =>
+        resource(resource3) { r3 =>
+          body(r1, r2, r3)
+        }
+      }
+    }
+
+  /** Performs an operation using four resources, and then releases the resources
+   * in reverse order, even if the operation throws an exception. This method
+   * behaves similarly to Java's try-with-resources.
+   *
+   * $suppressionBehavior
+   *
+   * @param resource1 the first resource
+   * @param resource2 the second resource
+   * @param resource3 the third resource
+   * @param resource4 the fourth resource
+   * @param body      the operation to perform using the resources
+   * @tparam R1 the type of the first resource
+   * @tparam R2 the type of the second resource
+   * @tparam R3 the type of the third resource
+   * @tparam R4 the type of the fourth resource
+   * @tparam A  the return type of the operation
+   * @return the result of the operation, if neither the operation nor
+   *         releasing the resources throws
+   */
+  def resources[R1: Releasable, R2: Releasable, R3: Releasable, R4: Releasable, A](
+      resource1: R1,
+      resource2: => R2,
+      resource3: => R3,
+      resource4: => R4
+  )(body: (R1, R2, R3, R4) => A): A =
+    resource(resource1) { r1 =>
+      resource(resource2) { r2 =>
+        resource(resource3) { r3 =>
+          resource(resource4) { r4 =>
+            body(r1, r2, r3, r4)
+          }
+        }
+      }
+    }
+
+  /** A typeclass describing how to release a particular type of resource.
+   *
+   * A resource is anything which needs to be released, closed, or otherwise cleaned up
+   * in some way after it is finished being used, and for which waiting for the object's
+   * garbage collection to be cleaned up would be unacceptable. For example, an instance of
+   * [[java.io.OutputStream]] would be considered a resource, because it is important to close
+   * the stream after it is finished being used.
+   *
+   * An instance of `Releasable` is needed in order to automatically manage a resource
+   * with [[Using `Using`]]. An implicit instance is provided for all types extending
+   * [[java.lang.AutoCloseable]].
+   *
+   * @tparam R the type of the resource
+   */
+  trait Releasable[-R] {
+
+    /** Releases the specified resource. */
+    def release(resource: R): Unit
+  }
+
+  object Releasable {
+
+    /** An implicit `Releasable` for [[java.lang.AutoCloseable `AutoCloseable`s]]. */
+    implicit object AutoCloseableIsReleasable extends Releasable[AutoCloseable] {
+      def release(resource: AutoCloseable): Unit = resource.close()
+    }
+
+    /** An implicit `Releasable` for [[scala.io.Source `Source`s]] which aren't [[java.lang.AutoCloseable `AutoCloseable`s]] in Scala 2.11. */
+    implicit object SourceReleasable extends Releasable[Source] {
+      def release(resource: Source): Unit = resource.close()
+    }
+  }
+
```

Comment:
guess we can remove this empty line

Reply:
It was there before, so I left it in.  Can it be configured in scalafmt if it is a problem?

Reply:
it's fine to clean up formatting in the course of making changes, but it's also fine not to